src/rapids_pre_commit_hooks/lint.py

```diff
@@ -117,31 +120,52 @@ def print_warnings(self, fix_applied=False):
 
             for replacement in warning.replacements:
                 line_index = self.line_for_pos(replacement.pos[0])
+                newtext = replacement.newtext
+                if match := self.NEWLINE_RE.search(newtext):
+                    newtext = newtext[: match.start()] + "..."
+                    long = True
+                else:
+                    long = False
+
                 print(f"In file {self.filename}:{line_index + 1}:")
-                self.print_highlighted_code(replacement.pos, replacement.newtext)
+                self.print_highlighted_code(replacement.pos, newtext)
                 if fix_applied:
-                    print("note: suggested fix applied")
+                    if long:
+                        print("note: suggested fix applied but is too long to display")
+                    else:
+                        print("note: suggested fix applied")
                 else:
-                    print("note: suggested fix")
+                    if long:
+                        print(
+                            "note: suggested fix is too long to display, use --fix to "
+                            "apply it"
+                        )
+                    else:
+                        print("note: suggested fix")
                 print()
 
     def print_highlighted_code(self, pos, replacement=""):
         line_index = self.line_for_pos(pos[0])
         line_pos = self.lines[line_index]
         left = pos[0] - line_pos[0]
 
+        if replacement:
+            replacement = f" {replacement}"
```

Comment:
Why add a space here? I think it's more confusing than the previous output.

Reply:
An alternative would be to write a diff/patch style output with - and +. That would also make the replacement line shorter, without all the ~~~~~~~~~~ characters.

Reply:
I’m fine with whatever you decide here.

Reply:
Diff-style output with `+` and `-` makes sense, but it wouldn't highlight the parts *within* the line that need to be changed (unless we did something fancy with terminal color output, which I'm open to doing.)

The `~~~` format was inspired by clang's warning output.

Reply:
I just checked out `rich` and agree that it look good for formatting the output. I'll use it to show a nicer diff.

Reply:
I've redone the output using `rich` and created a diff-like output for the replacements.